colossus/src/main/scala/colossus/core/WorkerManager.scala

```diff
@@ -1,84 +1,84 @@
-                                      ioSystem: IOSystem,
-                                      workerFactory: WorkerFactory)
+private[colossus] class WorkerManager(workerRefs: IOSystem.WorkerRefs, ioSystem: IOSystem, workerFactory: WorkerFactory)
     extends Actor
     with ColossusLogging
     with Stash {
+
   import WorkerManager._
-  import akka.actor.OneForOneStrategy
-  import akka.actor.SupervisorStrategy._
-  import context.dispatcher
 
-  import ioSystem.numWorkers
+  import context.dispatcher
 
   /*
    * This strategy is chosen specifically for durability.  We don't want to kill the entire system if workers start
    * excepting.
    */
-  override val supervisorStrategy = {
+  override val supervisorStrategy: OneForOneStrategy = {
     OneForOneStrategy() {
-      case _: Exception => Restart
+      case e: Exception =>
+        // TODO default is to restart on Exception... adding logging to see if this code gets hit
+        error(s"[${ioSystem.name}] Supervisor strategy error $e")
+        Restart
     }
   }
 
-  val workers = (1 to numWorkers).map { i =>
+  val workers: Seq[ActorRef] = (1 to ioSystem.numWorkers).map { i =>
     workerFactory.createWorker(i, ioSystem, context)
   }
 
-  val workerRouter =
-    context.actorOf(Props.empty.withRouter(RoundRobinGroup(Iterable(workers.map(_.path.toString): _*))))
-  var registeredServers = collection.mutable.ArrayBuffer[ServerRef]()
-
-  //this is used when the manager receives a Connect request to round-robin across workers
-  var nextConnectIndex = 0
+  val workerRouter: ActorRef = context.actorOf(
+    Props.empty.withRouter(RoundRobinGroup(Iterable(workers.map(_.path.toString): _*)))
+  )
 
-  var latestSummary: Seq[ConnectionSnapshot] = Nil
-  var latestSummaryTime                      = 0L
+  private var registeredServers         = ArrayBuffer.empty[ServerRef]
+  private var latestSummary             = Seq.empty[ConnectionSnapshot]
+  private var latestSummaryTime         = 0L
+  private var outstandingWorkerIdleAcks = 0
 
-  var outstandingWorkerIdleAcks = 0
-
-  def receive = waitForWorkers(Vector())
+  def receive: Receive = waitForWorkers(Vector())
 
   def waitForWorkers(ready: Vector[WorkerRef]): Receive = {
-    case ReadyCheck => {
+    case ReadyCheck =>
       sender ! WorkersNotReady
-    }
-    case WorkerReady(worker) => {
+
+    case WorkerReady(worker) =>
       val nowReady = ready :+ worker
-      if (nowReady.size == numWorkers) {
-        info("All Workers reports ready, lets do this")
-        workerAgent.set(nowReady)
+      if (nowReady.size == ioSystem.numWorkers) {
+        info(s"[${ioSystem.name}] All Workers reports ready, lets do this")
+        workerRefs.set(nowReady)
         context.system.scheduler.scheduleOnce(IdleCheckFrequency, self, IdleCheck)
         unstashAll()
         context.become(running)
       } else {
         context.become(waitForWorkers(nowReady))
       }
-    }
-    case other => stash()
+
+    case _ =>
```

Comment:
don't we want to keep a reference to this message and log it?

Reply:
Don't think so, at least we aren't at the moment. Everything will be unstashed later.